src/sas/qtgui/Calculators/GenericScatteringCalculator.py

```diff
@@ -270,19 +624,26 @@ def complete_loading(self, data=None):
             logging.info(log_msg)
             raise
         logging.info("Load Complete")
-        self.cmdLoad.setEnabled(True)
-        self.cmdLoad.setText('Load')
-        self.cmdCompute.setEnabled(True)
-        self.cmdCompute.setText('Compute')
-        self.trigger_plot_3d.emit()
+        # Once data files are loaded allow them to be enabled
+        if load_nuc:
+            self.checkboxNucData.setEnabled(True)
+        else:
+            self.checkboxMagData.setEnabled(True)
+        # update GUI if these files are already enabled
+        if (load_nuc and self.is_nuc) or ((not load_nuc) and self.is_mag):
+            self.update_gui()
+        # reset verification now we have loaded new files
+        self.verification_occurred = False
+        self.verified = False
+        self.verify_files_match()
```

Comment:
Shouldn't loading a file cause checking the corresponding checkbox as well? (`checkboxNucData` / `checkboxMagData`)
As a user I would probably expect the dialog to be updated with all the information read from the file in a state ready for calculation.

Reply:
Another comment/observation
- load some nuclear data
- Total volume, Nodes data and Step size data remains unchanged (because the checkbox is unchecked)
- click on the nuclear data checkbox to activate the dataset
- Total volume udpates, so do Nodes/Step size
- ucheck the checkbox
- the previous (original) state of Total Volume, Nodes and Step Size does not come back but instead we get red NaNs, which, again is not very user friendly, considering the user didn't do anything that would warrant the errors.

Reply:
Loading a file also enables it as of 04bff3baf1baac245e69906b3e0197024a7b0a56. It definitely feels much smoother to use now.

Reply:
Regarding the second point I'm not entirely sure what the most ideal behaviour is. Currently all values are left alone when files are enabled or disabled, which as you have pointed out leaves a rather ugly case where large amounts of the GUI become invalid for what should be a fairly basic operation. The two behaviours I see with a change are:

- what should the values be reset to - I would be inclined to suggest the last values that were present when no files were selected, however I can also see an argument that always reverting back to the default values might be more predictable behaviour for the user
 - Which textboxes should be reset - certainly nodes and stepsize, which also resets the total volume, however I think a case could be made that if those textboxes change the nuclear and magnetic slds should also revert - rather than remaining on the values which describe a no-longer-enabled file. Personally I would think that the up_frac_in, up_theta, etc. should remain unchanged - as they describe the configuration, rather than the material, but then we have two different behaviours for textboxes (across different panels) when files are disabled

I will try to have a play and see what feels most natural, do you have any thoughts?

Reply:
1 - I think unselecting the checkbox means "unloading" of the data, so the default values should replace the read-in numbers. Showing last values (pre-load) is probably less intuitive and surely much more complex programatically as you have to keep track of state changes.
2 - This should be answered by someone familiar with the functionality and user expectations. Maybe this can be raised in an email to the devs?

Reply:
After speaking with Dirk Honecker regarding this I've implemented a reasonable first solution to this which feels relatively smooth to use in 07d4bdacb59fc5c18be7229277a379163349104d

Any sample dependent parameters (those within the `SLD Pixel Info` panel and the total volume) are reset to their default values after both files are deselected. Any instrument dependent properties (all other panels) are left as is.

In ab6a404e8b1a750715edb52cfe962301744e5f73 I've also added code so that when nuclear/magnetic files are loaded magnetic/nuclear SLDs are respectively set to 0. That is to say enabling/disabling a file always sets the user editable options to their defaults. This also prevents the annoying situation where on loading a magnetic file into the initial setup the starting nuclear sld was non-zero (left over from the default starting value) - so unless the user made further alterations the file they had loaded in would not be computed 'as is'.

Reply:
this looks good!